### PR TITLE
Add rightclick options plugin

### PR DESCRIPTION
**Pull request by widget-**
Adds a plugin that can

* Highlight user-defined right-click menu items

* Dim user-defined right-click menu items

* Hide some pre-selected menu items:

   * Examine
   * Trade with
   * Follow
   * Withdraw-5
   * Withdraw-10
   * Withdraw-All-But-One
   * Deposit worn items

Example:

image

### CODE REVIEW

**Comment by raiyni** [comment]
That sounds like something on the rejected list because you can hide down to withdraw-x.

**Comment by MagicfTail** [comment]
Wouldn't hiding all unwanted options make all the menu swaps from https://github.com/runelite/runelite/wiki/Rejected-or-Rolled-Back-Features possible?

**Comment by widget-** [comment]
Ah, yep, you're right. CONTRIBUTING.md should really have a link to that page.

**Comment by widget-** [comment]
Though I'd like to see more thought given to #1647 as mobile supports it, I'm fine with changing this to a large set of checkboxes for removing common annoyances (examine, trade, follow, deposit worn items, etc.)

**Comment by raiyni** [comment]
Jagex said they would consider qol things that mobile provides. Until then, it's still less effort to right click and withdraw than it is to use the bank on mobile.

**Comment by deathbeam** [comment]
If you want this added, you should probably remove the hiding of menu entries part

**Comment by widget-** [comment]
I've updated the code and the original comment with some menu removing but nothing which conflict with the rejected or rolled back features page.

File: runelite-client/src/main/java/net/runelite/client/plugins/rightclick/RightclickOptionsConfig.java [hunk 1]
```diff
@@ -0,0 +1,170 @@
...
+		return Color.decode("#808080");
+	}
+
+	@ConfigItem(
+		keyName = "hideExamine",
+		name = "Hide \"Examine\"",
+		description = "Hides Examine from all right-click menus",
+		position = 5
+	)
+	default boolean hideExamine()
```
Comment (raiyni) [inline runelite-client/src/main/java/net/runelite/client/plugins/rightclick/RightclickOptionsConfig.java]: Having all of these hides can be abused.

File: runelite-client/src/main/java/net/runelite/client/plugins/rightclick/RightclickOptionsConfig.java [hunk 2]
```diff
@@ -0,0 +1,170 @@
...
+		return false;
+	}
+
+	@ConfigItem(
+		keyName = "hideWithdraw5",
+		name = "Hide \"Withdraw-5\"",
+		description = "Hides Withdraw-5 from all right-click menus",
+		position = 8
+	)
+	default boolean hideWithdraw5()
```
Comment (raiyni) [inline runelite-client/src/main/java/net/runelite/client/plugins/rightclick/RightclickOptionsConfig.java]: This is just a way around not swapping withdraw-x

Comment (widget-) [reply runelite-client/src/main/java/net/runelite/client/plugins/rightclick/RightclickOptionsConfig.java]: It makes it the second option.

File: runelite-client/src/main/java/net/runelite/client/plugins/rightclick/RightclickOptionsPlugin.java [hunk 3]
```diff
@@ -0,0 +1,222 @@
...
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+package net.runelite.client.plugins.rightclick;
+
+import com.google.common.base.Splitter;
+import com.google.common.eventbus.Subscribe;
+import com.google.inject.Provides;
+import lombok.Getter;
+import net.runelite.api.*;
```
Comment (raiyni) [inline runelite-client/src/main/java/net/runelite/client/plugins/rightclick/RightclickOptionsPlugin.java]: No star imports except for statics

File: runelite-client/src/main/java/net/runelite/client/plugins/rightclick/RightclickOptionsPlugin.java [hunk 4]
```diff
@@ -0,0 +1,222 @@
...
+import net.runelite.api.events.*;
```
Comment (raiyni) [inline runelite-client/src/main/java/net/runelite/client/plugins/rightclick/RightclickOptionsPlugin.java]: No star imports except for statics

File: runelite-client/src/main/java/net/runelite/client/plugins/rightclick/RightclickOptionsPlugin.java [hunk 5]
```diff
@@ -0,0 +1,222 @@
...
+import net.runelite.client.config.ConfigManager;
+import net.runelite.client.plugins.Plugin;
+import net.runelite.client.plugins.PluginDescriptor;
+import net.runelite.client.plugins.rightclick.config.MenuHighlightMode;
+import net.runelite.client.util.ColorUtil;
+import net.runelite.client.util.WildcardMatcher;
+
+import javax.inject.Inject;
+import java.awt.*;
```
Comment (raiyni) [inline runelite-client/src/main/java/net/runelite/client/plugins/rightclick/RightclickOptionsPlugin.java]: No star imports except for statics

File: runelite-client/src/main/java/net/runelite/client/plugins/rightclick/RightclickOptionsConfig.java (hunk 2 above)
Comment (Abextm) [reply runelite-client/src/main/java/net/runelite/client/plugins/rightclick/RightclickOptionsConfig.java]: I would like to not touch the bank menu at all.

**Comment by apicard14** [comment]
im trying to to make "Deposit-All" the first item option while banking instead of "Deposit-1" can anyone help me with that? or a way to shift click to deposit all of 1 item

**Comment by Nightfirecat** [comment]
@apicard14 That can be done using the "All" button at the bottom of the bank interface, without using any Runelite features.

**Comment by apicard14** [comment]
@Nightfirecat i dont want to put the entire inventory in the bank, say i got 12 noted coal + various other items, i want to be able to single click/shift click the coal and it all goes in, not right click deposit all

**Comment by Nightfirecat** [comment]
@apicard14 It seems that the rejected feature mentioning this was removed at some point; after some discussion with the team, I've re-added this to the [Rejected features list](https://github.com/runelite/runelite/wiki/Rejected-or-Rolled-Back-Features):

> `Deposit X/Withdraw X` (with shift+click): Too powerful to implement. Jagex will implement if they want to

**Comment by abextm** [comment]
Jagex's [recent statement](https://secure.runescape.com/m=news/a=13/another-message-about-unofficial-clients?oldschool=1) prohibits a bunch of things related to removing menu entries in various scenarios.

**Comment by widget-** [comment]
Closed due to Jagex mandatory carpel tunnel announcement

**Pull request closed without merging**